apisix/plugins/ua-restriction.lua

```diff
@@ -142,7 +142,7 @@ function _M.access(conf, ctx)
         match = lrucache_useragent(user_agent, conf, match_user_agent, user_agent, conf)
     end
 
-    if match > MATCH_ALLOW then
+    if match > MATCH_ALLOW or (match == MATCH_NONE and conf.allowlist and not conf.denylist) then
```

Comment:
What happens if there is more than one UA？
It will depends on the last UA in the array and it looks strange

Reply:
```
   if type(user_agent) == "table" then
        for _, v in ipairs(user_agent) do
            if type(v) == "string" then
                match = lrucache_useragent(v, conf, match_user_agent, v, conf)
                if match > MATCH_ALLOW then
                    break
                end
            end
        end
    else
        match = lrucache_useragent(user_agent, conf, match_user_agent, user_agent, conf)
    end
```

When more than one UA, this code will traverse all of the UA talbe util `match > MATCH_ALLOW`, so even if `match == MATCH_NONE or match == MATCH_ALLOW` , it will still continue to traverse.

I don't know if I understand correctly, if not, can you give me a example to explain it?

Reply:
It's a bit complicated here, could we refactor this code?

Reply:
@monkeyDluffy6017 PLAT, thanks.

Reply:
@monkeyDluffy6017 PTAL, thanks.